### ISSUE TITLE
Can't build site because asset hashing is broken in Node >= 17

### ISSUE THREAD

**Issue by OyewoleOyedeji** [tailwindlabs/tailwindcss.com#1589]
I don't really know much about stuff like this but, when trying to build the documentation with `node>=16.20.0` it gives me this error

```log
./src/img/brand/tailwindcss-mark.svg
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:130:10)
    at getHashDigest (/home/oyedejioyewolemaxwell/Documents/Documentation/tailwindcss.com/node_modules/loader-utils/lib/getHashDigest.js:46:34)
    at /home/oyedejioyewolemaxwell/Documents/Documentation/tailwindcss.com/node_modules/loader-utils/lib/interpolateName.js:113:11
    at String.replace (<anonymous>)
    at interpolateName (/home/oyedejioyewolemaxwell/Documents/Documentation/tailwindcss.com/node_modules/loader-utils/lib/interpolateName.js:110:8)
    at Object.loader (/home/oyedejioyewolemaxwell/Documents/Documentation/tailwindcss.com/node_modules/file-loader/dist/index.js:29:48)
```

So, searching Google I found this question <https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported>, and the answer by David stated that an SSL vunerability was patched in versions greater than`16.19.0` and `npm audit fix --force` was the way to go in order for it to work (with breaking changes of course)

**Comment by OyewoleOyedeji** [comment]
I'll try and see what I can do to fix it (and possibly make it future-proof 😅)

**Comment by thecrypticace** [comment]
Hey! So this has to do with creating hashes for assets during build time. It's not a security vulnerability — it's an incompatibility with the version of Node you're using and the linked crypto library (e.g. OpenSSL). There was a breaking change in OpenSSL v3 with regards to initialization of some crypto routines and the error you see is a result of this. This happened when Node 17 was release as it was the first release that linked against OpenSSL 3.

The disabling of md4 in v3 was intentional because — in security contexts — it absolutely should not be used. However, for cache-busting it's not actually significant problem. But we should still update it.

**Comment by OyewoleOyedeji** [comment]
@thecrypticace thanks, I'm new to this kind of things

**Comment by thecrypticace** [comment]
I merged a fix for this in #1591. Should be capable of building all the way up to Node v20 now. Thanks! ✨
